File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        map<int,int>m;
        for(auto it:nums){
            m[it]++;
        }
        vector<int>v;
        for(auto it:m){
            if(it.second==2)v.push_back(it.first);
        }
        return v;
        
    }
};

File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        Map<Integer,Integer>m=new HashMap();
        List<Integer>list=new ArrayList<>();

        for(int n:nums){
            if(m.containsKey(n))list.add(n);
            m.put(n,1);
        }

        return list;
    }
}